warehouse/csp.py

```diff
@@ -32,13 +31,15 @@ def content_security_policy_tween(request):
 
 
 def csp_factory(_, request):
-    return copy.deepcopy(request.registry.settings.get("csp", {}))
+    return collections.defaultdict(
+        list, request.registry.settings.get("csp", {})
+    )
 
 
 def includeme(config):
     config.register_service_factory(csp_factory, name="csp")
     # Enable a Content Security Policy
-    config.add_settings({
+    config.add_settings(collections.defaultdict(list, {
```

Comment:
I take that back. the `defaultdict` doesn't hold when applied to `request.registry.settings` because of the `update`. So no, this doesn't have to be a `defaultdict` (will change it), but I'll need to update the factory function.